### PR TITLE
Fix : changed the relationship between Pharmacy and  the Pharmacy_user

### PR DESCRIPTION
**Pull request by Kavindu-248**

### CODE REVIEW

File: apps/dispensary/views.py [hunk 1]
```diff
@@ -21,6 +21,19 @@ def retrieve(self, request, pk=None):
 
 
 
+class PharmacyViewSet(viewsets.ModelViewSet):
```
Comment (Raam124) [inline apps/dispensary/views.py:24]: we can't have a create method for pharmacy add a function to create a pharmacy automatically when a pharmacy user is registering

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/dispensary/views.py (hunk 1 above)
Comment (Raam124) [reply apps/dispensary/views.py:24]: only need tt have a get and update endpoints here

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/dispensary/views.py (hunk 1 above)
Comment (Kavindu-248) [reply apps/dispensary/views.py:24]: Updated the necessary changes

File: apps/dispensary/views.py [hunk 2]
```diff
@@ -20,6 +21,23 @@ def retrieve(self, request, pk=None):
         return Response(serializer.data)
 
 
+class PharmacyViewSet(viewsets.ModelViewSet):
```
Comment (Raam124) [inline apps/dispensary/views.py:24]: the modelviewset will automatically enable all the method better use the viewset here

File: apps/dispensary/views.py [hunk 3]
```diff
@@ -20,6 +21,23 @@ def retrieve(self, request, pk=None):
...
+    queryset = Pharmacy.objects.all()
+    serializer_class = PharmacySerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def perform_update(self, serializer):
+        serializer.save(pharmacy_user=self.request.user.pharmacyuser)
+
+
```
Comment (Raam124) [inline apps/dispensary/views.py:32]: do not leave white space like this

File: apps/users/serializers.py [hunk 4]
```diff
@@ -6,26 +6,30 @@
...
 class AuthRegisterSerializer(serializers.ModelSerializer):
     confirm_password = serializers.CharField(required=True, write_only=True, min_length=6)
+    pharmacy_name =serializers.CharField(required=False)
+    address = serializers.CharField(required=False)
+
     class Meta:
         model = get_user_model()
         fields = ['id', 'first_name', 'last_name',
-                  'email', 'password', 'confirm_password', 'phone', 'role']
+                  'email', 'password', 'confirm_password', 'phone', 'role', 'pharmacy_name', 'address']
```
Comment (Raam124) [inline apps/users/serializers.py:32]: do not get address here just keep it null user can update it later

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/dispensary/views.py (hunk 2 above)
Comment (Kavindu-248) [reply apps/dispensary/views.py]: Changed to ViewSet custom list and update methods

File: apps/dispensary/views.py (hunk 3 above)
Comment (Kavindu-248) [reply apps/dispensary/views.py:43]: noted.

File: apps/users/serializers.py (hunk 4 above)
Comment (Kavindu-248) [reply apps/users/serializers.py]: Removed address from the fields, up-to user to update the data.

File: apps/dispensary/views.py [hunk 5]
```diff
@@ -20,6 +20,35 @@ def retrieve(self, request, pk=None):
...
+class PharmacyViewSet(viewsets.ViewSet):
+
+    def list(self, request):
+        queryset = Pharmacy.objects.all()
+        serializer = PharmacySerializer(queryset, many=True)
+        return Response(serializer.data)
+
+    def update(self, request, pk=None):
+        pharmacy = get_object_or_404(Pharmacy, pk=pk)
+        data={
```
Comment (Raam124) [inline apps/dispensary/views.py:32]: directly access the data do not have to declare a dict again here

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/dispensary/views.py (hunk 5 above)
Comment (Kavindu-248) [reply apps/dispensary/views.py]: removed the dictionary

**Review by Raam124 — APPROVED** [review]

**Pull request merged**